src/microbe_stage/editor/PatchMapNode.cs

```diff
@@ -276,29 +279,32 @@ public void UpdateVisibility()
         }
     }
 
-    public void ShowEventVisuals(IReadOnlyList<WorldEffectVisuals> list)
+    public void ShowEventVisuals(IReadOnlyList<WorldEffectVisuals.WorldEffectTypes> list)
     {
-        eruptionEventIndicator.Visible = false;
-        glaciationEventIndicator.Visible = false;
+        if (eventIconsContainer == null)
+            return;
 
-        var count = list.Count;
+        eventIconsContainer.QueueFreeChildren(false);
 
-        for (int i = 0; i < count; ++i)
+        foreach (var effectType in list)
         {
-            switch (list[i])
+            var eventIcon = eventIconScene.Instantiate<PatchMapEventIcon>();
+
+            if (!WorldEffectVisuals.EventsTooltips.TryGetValue(effectType, out var tooltipText))
+            {
+                throw new Exception($"Missing tooltip for {effectType}");
+            }
+
+            eventIcon.TooltipText = Localization.Translate(tooltipText);
```

Comment:
I can't force the localization tool to include these tooltip texts :/

Reply:
The translation extraction doesn't run the code. It uses just a syntax tree parser, so it can only see `.Translate("STRING_LITERAL")` style patterns. The tooltip text here is a variable so the translation extraction has no clue what it should extract here. You need to put the Localization.Translate call to where the actual string literal is (I'm assuming, I haven't read the code above this yet).